Source/UI/CommonUI/View/Component/InputView.swift

```diff
@@ -216,9 +222,12 @@ extension InputView: TextFieldViewDelegate {
     func textFieldShouldEndEditing(textField: UITextField, replacementString: String) -> Bool {
         let shouldEndEditing = delegate?.textFieldShouldEndEditing(textField: textField, replacementString: replacementString) ?? true
         if shouldEndEditing {
-            textFieldContainerBorder.layer.borderColor = (style?.error?.isHidden ?? true) ?
-            style?.textfield.borderStyle.normalColor.cgColor :
-                style?.textfield.borderStyle.errorColor.cgColor
+            if let color = (style?.error?.isHidden ?? true) ?
+            style?.textfield.borderStyle.normalColor :
+                    style?.textfield.borderStyle.errorColor {
+                textFieldContainerBorder.updateBorderColor(to: color)
+            }
```

Comment:
I don't think we should use ternary operator in an if statement. It makes things very fuzzy.

ie:
```swift
if let variable = condition ? option1 : option2 {
```

Your concern seems to be strictly on the optionality of style, so if you just unwrap that, it should become a lot easier.

```swift
if shouldEndEditing,
    let style = style {
        let color = style.error?.isHidden == false : style.textfield.borderStyle.errorColor : style.textfield.borderStyle.normalColor
        textFieldContainerBorder.updateBorderColor(to: color)
}
```